app/src/full/java/io/homeassistant/companion/android/settings/SettingsWearViewModel.kt

```diff
@@ -0,0 +1,117 @@
+package io.homeassistant.companion.android.settings
+
+import android.app.Activity
+import android.net.Uri
+import android.util.Log
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.tasks.Tasks
+import com.google.android.gms.wearable.CapabilityClient
+import com.google.android.gms.wearable.DataItem
+import com.google.android.gms.wearable.DataMap
+import com.google.android.gms.wearable.DataMapItem
+import com.google.android.gms.wearable.PutDataMapRequest
+import com.google.android.gms.wearable.Wearable
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import kotlinx.coroutines.launch
+
+class SettingsWearViewModel : ViewModel() {
+
+    private lateinit var integrationUseCase: IntegrationRepository
+
+    private val TAG = "SettingsWearViewModel"
+    private val CAPABILITY_WEAR_FAVORITES = "send_home_favorites"
+
+    var entities = mutableStateMapOf<String, Entity<*>>()
+        private set
+    var favoriteEntityIds = mutableStateListOf<String>()
+
+    fun init(integrationUseCase: IntegrationRepository) {
+        this.integrationUseCase = integrationUseCase

```

Comment:
I think you should be able to inject this?

Reply:
So I wasnt able to I thought we had to do something similar to the comment in the Wear view model so I mimiced what we did for now.

Reply:
https://developer.android.com/training/dependency-injection/hilt-jetpack#viewmodels